src/main.py

```diff
@@ -210,7 +210,7 @@ def build(self):
         ModelManagerSingleton(m)
 
         # initialise GRBLSettingsManagerSingleton with machine to manage GRBL settings
-        GRBLSettingsManagerSingleton(m)
+        self.grbl_settings_manager = GRBLSettingsManagerSingleton(m)
```

Comment:
Move this up to the class definitions so it can be accessed via App.get_running_app() (At some point I want to get rid of the singletons again. I didn't realize they are that slow in python.)

Reply:
The class can still be accessed even if it's left in this place. See call_grbl_overwrite() in popups.py